### ISSUE TITLE
REST API listCatalogs endpoint does not use parameters page_token and max_results

### ISSUE THREAD

**Issue by VillePuuska** [unitycatalog/unitycatalog#53]
**Describe the bug**

When calling the Unity Catalog REST API endpoint [listCatalogs](https://github.com/unitycatalog/unitycatalog/blob/main/api/Apis/CatalogsApi.md#listcatalogs), the parameters `page_token` and `max_results` specified in the API spec aren't used.

**To Reproduce**

Python example. With Unity Catalog running locally with more than one catalog created, running
```python
import requests

print(
    requests.get(
        "http://localhost:8080/api/2.1/unity-catalog/catalogs",
        params={
            "max_results": 1,
        },
    ).json()
)
```
prints all catalogs, not just 1, and `next_page_token` is null. For comparison, running
```python
import requests

print(
    requests.get(
        "http://localhost:8080/api/2.1/unity-catalog/tables",
        params={
            "catalog_name": "unity",
            "schema_name": "default",
            "max_results": 1,
        },
    ).json()
)
```
only prints 1 table and `next_page_token` is not null.

**Expected behavior**

When setting `max_results` parameter, don't return all catalogs and specify `next_page_token`.

**System [please complete the following information]:**

-   OS: Ubuntu 20.04.6 (GitHub Codespaces, default image)

**Additional context**

I'm not a Java dev, but I'm guessing the problem stems from
- the handler not getting the parameters here: https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/service/CatalogService.java#L27
- and the `listCatalogs` method of `CatalogRepository` class not taking the parameters here: https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java#L55

**Comment by Datt1234** [comment]
In

**Comment by VillePuuska** [comment]
Was fixed by #292
